components/widgets/menu/menu_items/submenu_item_action.jsx

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import * as Utils from 'utils/utils.jsx';
+import {ModalIdentifiers} from 'utils/constants';
+import SubMenuModal from '../../../submenu_modal/submenu_modal.jsx';
+
+export default class SubMenuItemAction extends React.PureComponent {
```

Comment:
Given that this component is the actual full implementation of the subitem not only the action part (unlike the `MenuItem` and `MenuItemAction` composition), I suggest renaming this to`SubMenuItem`.  

Also, something to think about if there's any value in a refactoring this component to try to reuse the `MenuItem` and extend it to a `SubMenuItem` or `SubmenuItemAction` of any sort.

Reply:
Name changed to SubMenuItem

Originally, SubmenuItem action used MenuItem, however over time it seemed easier to encapsulate the functionality all inside of SubMenuItem instead.

Do feel strongly SubMenuItem should extend MenuItem?  I am not opposed to a refactor, though I am not sure what is gained by doing so in this case.  Thoughts?

Reply:
Do not feel strongly about it. Just a thought if it was something that made sense to keep the  architecture of the current menu definitions going. If there is no value to it then no need to force it 👍